fullrt/dht.go

```diff
@@ -334,36 +329,20 @@ func (dht *FullRT) runCrawler(ctx context.Context) {
 		addrs = append(addrs, dht.bootstrapPeers...)
 		dht.peerAddrsLk.Unlock()
 
-		for k := range foundPeers {
-			delete(foundPeers, k)
-		}
+		clear(foundPeers)
 
 		start := time.Now()
 		limitErrOnce := sync.Once{}
 		dht.crawler.Run(ctx, addrs,
 			func(p peer.ID, rtPeers []*peer.AddrInfo) {
-				conns := dht.h.Network().ConnsToPeer(p)
-				var addrs []multiaddr.Multiaddr
-				for _, conn := range conns {
-					addr := conn.RemoteMultiaddr()
```

Comment:
we don't want to record the current connect addresses since remote nodes could be using upnp, and may not use the same address in the future.

we want to record the listen addresses they share with us using identify, which are stored in the host's peerstore.